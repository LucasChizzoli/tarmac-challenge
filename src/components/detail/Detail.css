.detail-wrapper {
  text-align: center;
  padding: 20px;
}

.detail-image {
  height: 400px;
  width: 400px;
}

.member-name {
  font-size: 32px;
  font-weight: 500;
}

.member-role {
  font-size: 24px;
  font-weight: 700;
}