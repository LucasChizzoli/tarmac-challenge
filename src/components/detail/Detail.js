import React from 'react';
import './Detail.css';
import { PICTURE_URL } from '../../endpoints';
import tarmacTeam from '../../tarmac-people.json';
import PropTypes from 'prop-types'; 
import imageNotFound from '../../imageNotFound.jpg';

const Detail = (props) => {
  const index = Number(props.match.params.id);
  const member = tarmacTeam[index];
  return (
    <div className="detail-wrapper">
      <img alt={member.pic} src={`${PICTURE_URL}${member.pic}.png`} className="detail-image" onError={(e)=>{e.target.src=imageNotFound}}/>
      <div className="info-wrapper">
        <h3 className="member-name">{member.name}</h3>
        <h4 className="member-role">{member.role}</h4> 
      </div>
    </div>
  )
};

Detail.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    })
  })
}

export default Detail;