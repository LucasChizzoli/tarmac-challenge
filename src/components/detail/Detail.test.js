import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Detail from './Detail';
import '../../EnzymeSetup.js';

describe('<Detail />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const match = {
        params: {
          id:"0"
        }
      };
      const wrapper = shallow(<Detail match={match}/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});