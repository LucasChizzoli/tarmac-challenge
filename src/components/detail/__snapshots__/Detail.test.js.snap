// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> render() renders the component 1`] = `
<div
  className="detail-wrapper"
>
  <img
    alt="eugenia-diaz-arnesto"
    className="detail-image"
    onError={[Function]}
    src="https://tarmac.io/assets/members/eugenia-diaz-arnesto.png"
  />
  <div
    className="info-wrapper"
  >
    <h3
      className="member-name"
    >
      Eugenia DÃ­az Arnesto
    </h3>
    <h4
      className="member-role"
    >
      Developer
    </h4>
  </div>
</div>
`;
